From fcce34b13d66b8ca8dcf0179933be65c13cd5a28 Mon Sep 17 00:00:00 2001
From: Leon Schuermann <leon@is.currently.online>
Date: Wed, 17 Nov 2021 13:19:25 +0100
Subject: [PATCH 2/5] litex_sim: expose global simulation function symbols to
 modules

This is required for modules to be able to call globally defined
symbols to interact with other parts of the simulation, for example to
send inter-module messages, query the simulation time or request the
simulation to halt / resume.

Signed-off-by: Leon Schuermann <leon@is.currently.online>
---
 litex/build/sim/core/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/litex/build/sim/core/Makefile b/litex/build/sim/core/Makefile
index 585ef6e0..c7f45c56 100644
--- a/litex/build/sim/core/Makefile
+++ b/litex/build/sim/core/Makefile
@@ -4,11 +4,11 @@ UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Darwin)
 	CFLAGS += -I/usr/local/include/
 	LDFLAGS += -L/usr/local/lib
-	LDFLAGS += -lpthread -ljson-c -lz -lm -lstdc++ -ldl -levent
+	LDFLAGS += -lpthread -ljson-c -lz -lm -lstdc++ -ldl -levent -rdynamic
 else
 	CC ?= gcc
 	CFLAGS += -ggdb
-	LDFLAGS += -lpthread -Wl,--no-as-needed -ljson-c -lz -lm -lstdc++ -Wl,--no-as-needed -ldl -levent
+	LDFLAGS += -lpthread -Wl,--no-as-needed -ljson-c -lz -lm -lstdc++ -Wl,--no-as-needed -ldl -levent -rdynamic
 endif
 
 CFLAGS += -Wall -$(OPT_LEVEL) $(if $(COVERAGE), -DVM_COVERAGE) $(if $(TRACE_FST), -DTRACE_FST)
-- 
2.36.0

